extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/attribute/OTelDataAttribute.java

```diff
@@ -0,0 +1,53 @@
+package io.quarkus.vertx.http.runtime.attribute;
+
+import java.util.Map;
+
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Provide entries from the MDC section of the RoutingContext.
+ * This is especially helpful to put OTel data 'traceId' and 'spanId'
+ * into the access log.
+ */
+public class OTelDataAttribute implements ExchangeAttribute {
+
+    private final String dataKey;
+
+    public OTelDataAttribute(String dataKey) {
+        this.dataKey = dataKey;
+    }
+
+    @Override
+    public String readAttribute(RoutingContext exchange) {
+        Map<String, Object> mdcMap = exchange.vertx().getOrCreateContext().getLocal("io.quarkus.vertx.core.runtime.VertxMDC");
+        return (String) mdcMap.getOrDefault(dataKey, null);
```

Comment:
I know this is tricky. I was thinking this could go into the open-telemetry extension, but then `[ExchangeAttributeBuilder](https://github.com/quarkusio/quarkus/pull/27006#diff-287774a4f1c3b110b81bac31ef5eafa4372d24c820feb95563288be11238b421)` can't list it and I don't think there is something like a ServiceLoader mechanism available. Perhaps @stuartwdouglas knows?

Reply:
The service loader is available but not needed. 
We can try to bind this ExchangeAttribute like we do for the vert.x hooks. See: https://github.com/quarkusio/quarkus/blob/c6ed33903d9605467c22c530d00f8dd331c6dad0/extensions/opentelemetry/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/tracing/TracerProcessor.java#L196

Reply:
I think you should use `VertxMDC.INSTANCE` directly if you need to pass a Vert.x context , as it was done here https://github.com/quarkusio/quarkus/blob/main/extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java#L99 or call `MDC.get(dataKey));` since VertxMDC will be always there, and you can log everything in the MDC not only Otel stuff

Reply:
The thing is that only OTEL might be adding stuff to the MDC now, but the Vertx MDC implementation is in Vertx, and not in the OTEL extension, so I would really make this agnostic, and use the Vertx MDC instead of attaching it to OTEL

Reply:
> I think you should use `VertxMDC.INSTANCE` directly 

Great suggestion. I wasn't aware of it, but that makes life easier for everyone.

Reply:
> The thing is that only OTEL might be adding stuff to the MDC now, but the Vertx MDC implementation is in Vertx, and not in the OTEL extension, so I would really make this agnostic, and use the Vertx MDC instead of attaching it to OTEL

You are right. I am renaming this and also update documentation.

Reply:
The weird thing will be to see empty traceId and spanIds when the OTel extension is not used

Reply:
But that is also true for the console.log.format pictured here:

https://quarkus.io/guides/opentelemetry#create-the-configuration

Reply:
yeah... it can probably be improved